### ISSUE TITLE
CORS on printResume (turns out to be a Firebase memory issue)

### ISSUE THREAD

**Issue by smerth** [AmruthPillai/Reactive-Resume#310]
When I try to print a resume (single page resume or multi page resume) I get "loading..."  which never ends.   In the console I see a CORS error.  I know what CORS is, I don't know how to fix it in this app.  

The app is hosted on Firebase...  I think it has something to do with the functions...?

Any ideas, anyone?

**Comment by smerth** [comment]
When I view the logs for printResume on Firebase I get an "error memory limit exceeded" message.  I have read that sometimes Firebase / GCP functions report a CORS error when in fact there is an error in code..

<img width="1004" alt="Screen Shot 2020-08-25 at 15 18 19" src="https://user-images.githubusercontent.com/670090/91233527-40aa7180-e6e6-11ea-9564-1845b98c1553.png">

You can see in this log, I get 4 statements per function execution
- function starts
- finishes with status 204
- function starts
- memory limit exceeded

I don't know the function starts twice, I would expect it to start and then pass or fail...

Does anyone have a working printResume function (either on localhost or on firebase?)

Any ideas about how troubleshoots this?

**Comment by Radikad** [comment]
I'm also having the same issue 

Access to fetch at 'https://us-central1-react-resume-312db.cloudfunctions.net/printResume' from origin 'http://mydomain.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by Corrado2601** [comment]
@smerth: Give Firebase more memory ;) For this add to functions/index.js  `exports.printResume = functions
  .runWith({ memory: '2GB', timeoutSeconds: 30 })
  .https.onCall(async ({ id, type }, { auth }) => {`

**Comment by Radikad** [comment]
@Corrado2601 will it solve my problem too? but seems not for me.  😕

**Comment by Corrado2601** [comment]
@Radikad : I'll try ;) You have another problem with CORS (Different URL for consuming / transfer data).
Not the best security best practice but helpful. 

Add a Access-Control-Allow-Origin Header in the firebase.json 

`"hosting": [
    {
      "target": "temptonresume",
      "public": "public",
      "ignore": ["**/.*", "firebase.json", "**/node_modules/**"],
      "headers": [
        {
          "source": "**/*",
          "headers": [
            {
              "key": "cache-control",
              "value": "cache-control: public, max-age=86400, must-revalidate"
            }
          ]
        },

        **{
          "source": "**/*",
          "headers": [
            {
              "key": "Access-Control-Allow-Origin",
              "value": "*"
            }
          ]
        },**

        {
          "source": "static/**",
          "headers": [
            {
              "key": "cache-control",
              "value": "public, max-age=604800, immutable"
            }
          ]
        },
        {
          "source": "**/*.@(css|js|woff2|jpg|png)",
          "headers": [
            {
              "key": "cache-control",
              "value": "public, max-age=86400, immutable"
            }
          ]
        },
        {
          "source": "sw.js",
          "headers": [
            {
              "key": "cache-control",
              "value": "cache-control: public, max-age=86400, must-revalidate"
            }
          ]
        }
      ]
    }
  ],`

**Comment by Radikad** [comment]
Thank you @Corrado2601 but the error is still there. :(

**Comment by Corrado2601** [comment]
@Radikad : Have you deploy the project complete to firebase with functions, hosting, real database and so on ? This change is only for firebase. Not for testing in localhost envoirment !

**Comment by smerth** [comment]
Thanks @Corrado2601, that change worked for me!  And I learned a bit more about Firebase functions!


Will you make a pull request with the code change?  If not I will do it, since I think it will affect anyone who wants to launch the app on Firebase.

**Comment by smerth** [comment]
@Radikad, I am changing the title of the issue and closing it, because I think it is solved.  For other problems about deploying to Firebase maybe opening another issue is a good idea,... to keep things separate.
